src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebProxy.cs

```diff
@@ -4,62 +4,29 @@
 #nullable enable
 
 using System;
-using System.Net;
 
 namespace Microsoft.PowerShell.Commands
 {
-    internal class WebProxy : IWebProxy, IEquatable<WebProxy>
+    internal class WebProxy : System.Net.WebProxy, IEquatable<WebProxy>
```

Comment:
I would assume that r_eference equality_ is enough to make the cache work for most real-world scenarios. Then we could still remove this type.

Reply:
Using ReferenceEquals does not work, the tests fail

Reply:
I think it is still possible to remove the class - the Address property on `System.Net.WebProxy` is available and could be used to compare with the passed in Proxy parameter (if one is provided).

The Address property on System.Net.WebProxy is also settable, and can be changed without causing exceptions between requests from `HttpClient` so it may be possible to simply update the Address property on the existing Proxy provided:

* the WebSession already has a Proxy (i.e. this is not a brand new WebSession)
* the type of the Proxy is `System.Net.WebProxy`

Updating the proxy Address property does cause a new connection to be made (to connect to the new proxy), but this doesn't require creation of a new `HttpClientHandler`. 

Note that no new connection is required if the user updates the Address property to the same value it already had.

Reply:
> Using ReferenceEquals does not work, the tests fail

I don't see the commit.

Reply:
I tested in in local

Reply:
I see no reason why it wouldn't work. Unless there happens to be wrapping in PSObject which can be bypassed.

Reply:
It doesn't work because there is a test that checks that passing the same -Proxy <value> repeatedly with a persistent websession doesn't cause a new httpclienthandler to be created. If you use referenceequals then this will not be true.

The code from the Equals() could be moved into the CmdLet if you want to retain the existing behaviour but remove the internal WebProxy class completely.